### PR TITLE
test: Change Org

### PR DESCRIPTION
**Pull request by ShyamOOAI**
## Release Notes

- **New Features**
	- Introduced comprehensive UI testing for the Home, Ingestion, Login, Logs, Metrics, Traces, Pipelines, Rum, Dashboard, Alerts, Reports, Data, IAM, and About pages.
	- Added functionality to handle organization changes and validations across multiple pages, including default organization selections.
	- New test suite for validating organization change scenarios across various pages.
	- Enhanced login functionality for internal users.
	- Introduced new classes and methods for the Metrics, Streams, Rum, and Alerts pages to support UI testing.
	- Added new classes for the Data and IAM pages, expanding testing capabilities.
	- Enhanced Management page with navigation and validation capabilities.
	- New methods for validating page URLs and content across various pages.

- **Bug Fixes**
	- Improved validation methods to ensure proper navigation and content checks on various pages.

- **Documentation**
	- Updated testing framework to include new test cases and improved structure for organization change tests.

### CODE REVIEW

File: tests/ui-testing/pages/loginPage.js [hunk 1]
```diff
@@ -11,6 +11,13 @@ export class LoginPage {
     await this.page.goto(process.env["ZO_BASE_URL"]);
     console.log("ZO_BASE_URL", process.env["ZO_BASE_URL"]);
   }
+
+  async loginAsInternalUser() {
+    //await this.page.goto(process.env["ZO_BASE_URL"]);
+    await this.page.getByText('Login as internal user').click();
+    
+  }
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/loginPage.js:19]: updated as per suggestion

File: tests/ui-testing/pages/logsPage.js [hunk 2]
```diff
@@ -59,9 +59,28 @@ export class LogsPage {
...
   }
 
+  async logsPageDefaultMultiOrg() {
+
+    await this.page.locator('[data-test="navbar-organizations-select"]').getByText('arrow_drop_down').click();
+    await this.page.getByRole('option', { name: 'defaulttestmulti' }).locator('div').nth(2).click();
+   // await validateUrlContains(this.page, 'path');
+
+
+}
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/logsPage.js]: removed

File: tests/ui-testing/pages/logsPage.js [hunk 3]
```diff
@@ -59,9 +59,28 @@
...
+
+async logsPageURLValidation() {
+
+    await expect(this.page).toHaveURL(/defaulttestmulti/);
+
+}
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/logsPage.js:83]: similar

File: tests/ui-testing/pages/homePage.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    constructor(page) {
+        this.page = page;
+        this.homePageMenu = page.locator('[data-test="menu-link-\\/-item"]');
+
+    }
+    async gotoHomePage() {
+
+        await this.homePageMenu.click();
+
+    }
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/homePage.js:15]: as per application locator

File: tests/ui-testing/pages/managementPage .js [hunk 5]
```diff
@@ -24,6 +24,26 @@
...
+    async goToManagement() {
+        await this.page.waitForSelector('[data-test="menu-link-/settings/-item"]');
+        await this.managementMenuItem.click({ force: true });
+        await expect(this.page.getByRole('main')).toContainText('Management');
+    }
+
+    async managementPageDefaultMultiOrg() {
+        await this.page.locator('[data-test="navbar-organizations-select"]').getByText('arrow_drop_down').click();
+        await this.page.getByRole('option', { name: 'defaulttestmulti' }).locator('div').nth(2).click();
+    }
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/managementPage .js:36]: updated

File: tests/ui-testing/pages/managementPage .js [hunk 6]
```diff
@@ -24,6 +24,26 @@ export
         await this.managementMenuItem.click({ force: true });
     }
 
+    async goToManagement() {
+        await this.page.waitForSelector('[data-test="menu-link-/settings/-item"]');
+        await this.managementMenuItem.click({ force: true });
+        await expect(this.page.getByRole('main')).toContainText('Management');
+    }
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/managementPage .js:31]: to design as per steps and expected result

File: tests/ui-testing/playwright-tests/changeOrg.spec.js [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+import { TracesPage } from "../pages/tracesPage.js";
+import { RumPage } from "../pages/rumPage.js";
+import { PipelinesPage } from "../pages/pipelinesPage.js";
+import { DashboardPage } from "../pages/dashboardPage.js";
+import { StreamsPage } from "../pages/streamsPage.js";
+import { ReportsPage } from "../pages/reportsPage.js";
+import { AlertsPage } from "../pages/alertsPage.js";
+import { DataPage } from "../pages/dataPage.js";
+import { IamPage } from "../pages/iamPage.js";
+import { ManagementPage } from "../pages/managementPage .js";
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/playwright-tests/changeOrg.spec.js:16]: updated

File: tests/ui-testing/playwright-tests/changeOrg.spec.js [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+        await streamsPage.gotoStreamsPage();
+        await streamsPage.streamsPageURLValidation();
+  
+    });
+   
+
+   
+
+    
+});
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/playwright-tests/changeOrg.spec.js]: ok to design as per POM

File: tests/ui-testing/pages/streamsPage.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    constructor(page) {
+        this.page = page;
+        this.streamsPageMenu = page.locator('[data-test="menu-link-\\/streams-item"]');
+
+    }
+    async gotoStreamsPage() {
+
+        await this.streamsPageMenu.click();
+
+    }
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/pages/streamsPage.js:15]: similar

**Review by neha00290 — APPROVED** [review]

File: tests/ui-testing/playwright-tests/changeOrg.spec.js [hunk 10]
```diff
@@ -0,0 +1,328 @@
...
+
+        await logsPage.navigateToLogs();
+        await logsPage.validateLogsPage();
+    });
+
+    test("Logs Page change organisation validation", async ({ page }) => {
+
+        await logsPage.navigateToLogs();
+        await logsPage.logsPageDefaultMultiOrg();
+        await page.waitForTimeout(5000);
```
[the remark this answers is not in the document]
Comment (ShyamOOAI) [reply tests/ui-testing/playwright-tests/changeOrg.spec.js:79]: ok as per application

**Pull request merged**